woocommerce/woocommerce-gateway-stripe
# Adjust the displaying of OAuth connect buttons when not using SSL

## Changes proposed in this Pull Request:

The buttons to connect an account via OAuth only work when using a site with an SSL certificate. This can complicate testing on development environments or staging sites without SSL.

This PR:
- Returns the OAuth URL when in test mode even though the return URL doesn't use the `https` protocol, instead of returning an empty string
- On the settings page, displays a notice when the OAuth URL couldn't be retrieved, instead of displaying a button that doesn't trigger any action

## Testing instructions

1. Access your site without using SSL, be it using `localhost` or through JT/JN using `http`
2. Disconnect your account
  At the Settings tab > Account status > Three dots menu at the right > Disconnect, at `siteurl/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings`
3. Confirm there's a button that says "Create or connect a test account"

| Before | After |
|--------|--------|
| <img width="500" alt="image" src="https://github.com/user-attachments/assets/6ca50ecc-8416-4ef4-8538-d84d40149364"> | <img width="500" alt="image" src="https://github.com/user-attachments/assets/f3d7eae2-9737-4326-b43c-9a6b7b532142"> | 

4. Click on the button and complete the onboarding
5. Confirm that your test account is connected
6. Go to the Settings tab, at `siteurl/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings`
7. Under Account status, click on Configure connection
8. Under the Test tab, confirm that there's a button that says "Create or connect a test account"
  <img width="400" src="https://github.com/user-attachments/assets/27cf86b5-e78d-497a-b55c-18b379da1160">
9. Confirm you're redirected to the Stripe onboarding flow
10. Under the Live tab, confirm that no button is displayed and there's a notice instead
<img width="400" alt="image" src="https://github.com/user-attachments/assets/80b51672-fc5b-47d0-8a7e-c5f1a6081f1b">

**Test when the site has a SSL certificate**
1. Access your site through https, either via JT or JN
2. Go to the Settings tab > Account status > Configure connection, at `siteurl/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings`
3. Confirm that
  - A Connect button is displayed under both the Live and Test tabs
  - Clicking on the button on each tab takes you to the live and test onboarding flows, respectively
 4. Disconnect your account from the store, at the Settings tab > Account status > Three dots menu at the right > Disconnect
 5. Confirm that
   - There's a button for the live onboarding flow and a link for the test one
  - Clicking on the button on each tab takes you to the live and test onboarding flows, respectively
    <img width="400" alt="image" src="https://github.com/user-attachments/assets/c0ac9059-8057-420f-87c0-8d9886201f51">


---


**Post merge**
